ssp/krb5/authentifier.go

```diff
@@ -178,21 +198,44 @@ func (a *Authentifier) affirmLogin(ctx context.Context) error {
 }
 
 func (a *Authentifier) APRequest(ctx context.Context) ([]byte, error) {
-
 	cli, err := a.makeClient(ctx)
 	if err != nil {
 		return nil, fmt.Errorf("krb5: init: apreq: make client: %w", err)
 	}
 
 	a.client = cli
 
-	if err := a.affirmLogin(ctx); err != nil {
-		return nil, fmt.Errorf("krb5: init: apreq: affirm login: %w", err)
-	}
+	var (
```

Comment:
could you please hide the logic for ccacheWithoutTGT inside a.affirmLogin and a.getServiceTicket?

```
func (a *Authentifier) affirmLogin(...) {
    if ccacheWithoutTGT != nil {
        return nil
    }
    ...
}
```

func (a *Authentifier) getServiceTicket(sname string) (tkt, key, error) {

    if a.ccacheWithoutTGT != nil {
        c, ok := a.ccacheWithoutTGT.GetEntry(...)
        ...
        return tkt, key, nil
     }
     return a.client.GetServiceTicket(sname)
}
```